api/Infrastructure/Authentication/AuthenticationServiceCollectionExtension.cs

```diff
@@ -254,102 +254,155 @@ private static async Task OnPostAuthSuccess(
         {
             var judgeId = configuration.GetNonEmptyValue("PCSS:JudgeId");
             var homeLocationId = configuration.GetNonEmptyValue("PCSS:JudgeHomeLocationId");
-
-            // Default the user to his own external Judge Id.
-            if (!string.IsNullOrWhiteSpace(context.Principal.ExternalJudgeId()))
+            PersonSearchItem judge = null;
+            try
             {
-                var dashboardService = context.HttpContext.RequestServices.GetRequiredService<IDashboardService>();
-                var judges = await dashboardService.GetJudges();
-
-                // Find out the judge home location from list of judges
-                int.TryParse(context.Principal.ExternalJudgeId(), out int externalJudgeId);
+                if (int.TryParse(context.Principal.ExternalJudgeId(), out int externalJudgeId))
+                {
+                    judge = await GetJudgeById(context, externalJudgeId);
+                }
 
-                var judge = judges.FirstOrDefault(j => j.PersonId == externalJudgeId);
-                if (judge != null)
+                if (IsMongoDbConfigured(configuration))
                 {
-                    judgeId = externalJudgeId.ToString();
-                    homeLocationId = judge.HomeLocationId.ToString();
+                    var userDto = await HandleMongoDbUser(context, logger, claims);
```

Comment:
These chained calls should really be refactored into a service to support different user backends, but it doesn't really make sense to pay that cost for this story in particular.